### PR TITLE
fix: Donations settings bug(PATCH)

### LINKED ISSUE #5618: Donations PATCH error
**Describe the bug**
Donations aren't getting enabled on the admin panel. Throws a 409 conflict error.

**To Reproduce**
1. Go to 'Admin Panel'
2. Click on 'Modules Section'
3. See unexpected error

**Additional context**
Working on this.
screenshot from 2019-02-18 15-12-59

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #5618 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:

- Fixed the bug for donations where `before_patch `was checking for the wrong condition

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 @iamareebjamal Please review

**Comment by mrsaicharan1** [comment]
@iamareebjamal Shouldn't it be checking if `donations_include` is  False?

**Comment by uds5501** [comment]
@mrsaicharan1 Yes, it should be checking that. On the side note, I think you accidentally created your PR on a merged branch. (was it intended though?)

File: app/api/modules.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def before_patch(self, args, kwargs, data=None):
         :param data:
         :return:
         """
-        if data.get('donation_include'):
+        if not data.get('donation_include'):
```
Comment (iamareebjamal) [inline app/api/modules.py]: ???

This is wrong logic AFAIK

Comment (mrsaicharan1) [reply app/api/modules.py]: After these changes, I am able to turn the donations on. But turning them off, unexpected error occurs. There was vice-versa state before these changes. What could the error possibly be?

Comment (iamareebjamal) [reply app/api/modules.py]: You are checking if the data has donation_include or not. You should check if the settings has donations enabled

Comment (CosmicCoder96) [reply app/api/modules.py]: @mrsaicharan1 can you update this PR please.

Comment (mrsaicharan1) [reply app/api/modules.py]: @CosmicCoder96 Yes. Will update by today. Should I be checking for donations_include in `before_patch` itself?

Comment (CosmicCoder96) [reply app/api/modules.py]: >You are checking if the data has donation_include or not. You should check if the settings have donations enabled

Did you get the point he made? 
You checked if the request received contains a filed called donation_include. 
In addition to checking that, you also need to check if that field is present in the response, has its value as `true` or `false`. `data.get` only checks the existence of the field, `data['donation_include']` will give you the value of the field.

Comment (mrsaicharan1) [reply app/api/modules.py]: Got it. Will update it right now.

Comment (CosmicCoder96) [reply app/api/modules.py]: Also this check shouldnt be here, it should be present in the events api. If an event organizer tries to pass a value for donations, and they are not enabled in the system, an exception should be thrown.

Comment (mrsaicharan1) [reply app/api/modules.py]: @CosmicCoder96 Please tell me if there's anything else which is required

Comment (mrsaicharan1) [reply app/api/modules.py]: @CosmicCoder96 Can I get an update on this please?

Comment (mrsaicharan1) [reply app/api/modules.py]: @iamareebjamal Can you look into this as well?

File: app/api/modules.py [hunk 2]
```diff
@@ -29,9 +28,6 @@ def before_patch(self, args, kwargs, data=None):
         :param data:
         :return:
         """
-        if data.get('donation_include'):
```
Comment (iamareebjamal) [inline app/api/modules.py]: Why remove this

Comment (mrsaicharan1) [reply app/api/modules.py]: This Check should be present in the events api where if the donations are disabled and the organiser tries to pass something for donations, an exception should be thrown(which has already been incorporated in the code). As quoted by @CosmicCoder96, I think this check shouldn't be present here.

Comment (CosmicCoder96) [reply app/api/modules.py]: yeah @iamareebjamal I believe this exact check should be placed in the events api, and this very exception should be raised *after* checking from the modules api if donation is enabled in the system. If the admin has not enabled it, then this field should not be present in the request body at all.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 I believe that this check is already present in the events
API.

**Comment by mrsaicharan1** [comment]
@CosmicCoder96 I believe that this check is already present in the events
API.
https://github.com/fossasia/open-event-server/blob/development/app/api/events.py

**Comment by mrsaicharan1** [comment]
@iamareebjamal Please review

**Pull request merged**